client/blocks/eligibility-warnings/index.tsx

```diff
@@ -334,16 +336,20 @@ EligibilityWarnings.defaultProps = {
 const processMarketplaceExceptions = (
 	state: Record< string, unknown >,
 	eligibilityData: EligibilityData,
-	isEligible: boolean
+	isEligible: boolean,
+	isMarketplaceException?: boolean
 ) => {
```

Comment:
It's not very clear to me what a variable called `isMarketplaceException` means being passed to a method called `processMarketplaceExceptions`.

From what I'm seeing this value will be `true` every time a Marketplace Theme is passed in this flow, right? Why do we need this?

Reply:
If you follow `processMarketplaceExceptions`, you can see that it uses some logic that does not apply in the onboarding flow. Specifically, in the onboarding flow, the site could not have the feature `WPCOM_FEATURES_INSTALL_PURCHASED_PLUGINS` because is a free site about to be upgraded **but** we still need to filter out `NO_BUSINESS_PLAN` from the eligibility holds.

The rationale for this code is:  Call `processMarketplaceExceptions`, but in this scenario, I can ensure that is a Marketplace exception because it is a Marketplace product. The way to indicate that is with this `isMarketplaceException ` flag.

> Why do we need this?

If we don't have this logic or an alternative, for free sites, you would see incorrect eligibility warnings like 
<img width="550" alt="CleanShot 2023-09-15 at 12 31 36@2x" src="https://github.com/Automattic/wp-calypso/assets/3519124/7c8c25d2-f00e-4d8b-800d-32aa4952162e">

As an alternative that I have found in other parts of the code is to copy this logic 
```
	eligibilityData.eligibilityHolds = eligibilityData.eligibilityHolds.filter(
		( hold ) => hold !== 'NO_BUSINESS_PLAN'
	);
```
 before calling the `EligibilityWarnings` component, but I don't think is better than this flag. Please let me know if you see better alternatives for this.

Reply:
I have reworded the `isMarketplaceException` prop to `isOnboarding` and I have changed the logic a bit to clarify the intention. Now, in `processMarketplaceExceptions` there could be two potential flows:
- Previous flow where it checks the site features
- New flow where, if it is the `isOnboarding` flow, we are not interested in the site features, we need to filter the `eligibilityHolds`

Commit here https://github.com/Automattic/wp-calypso/pull/81412/commits/a599916f39b5d7b65882cbca80279f0bae30b191. Please, let me know if it improves the clarity

Reply:
Yes, this looks much better now, thanks for the change.